src/vos/vos_io.c

```diff
@@ -1252,53 +1269,142 @@ vos_update_end(daos_handle_t ioh, uint32_t pm_ver, daos_key_t *dkey, int err)
 	err = vos_publish_blocks(ioc->ic_obj, &ioc->ic_blk_exts, true);
 
 abort:
+	if (dth != NULL) {
+		if (err == 0) {
+			err = vos_dtx_prepared(dth);
+		} else if (err == -DER_INPROGRESS &&
+			   dth->dth_flags & DTX_F_AOC &&
+			   !daos_is_zero_dti(&dth->dth_conflict)) {
+			/* XXX: The leader modification has been done,
+			 *	but current replica failed to execute
+			 *	the modification because of some uncommitted
+			 *	DTX, then it should be the case that we have
+			 *	ever failed to abort such DTX on this replica.
+			 *	Let's force abort such conflict uncommitted)
+			 *	DTX and retry.
+			 *
+			 *	Abort current PMDK transaction before retry.
+			 */
+			umem_tx_abort(umem, err);
+			goto again;
```

Comment:
The ic_actv_at is already being cleared to zero above, this 'goto again' will miss scm publishing.

Reply:
It is resolved in the push.